src/app/modules/item/pages/group-progress-grid/group-progress-grid.component.ts

```diff
@@ -259,4 +264,37 @@ export class GroupProgressGridComponent implements OnChanges, OnDestroy {
       });
   }
 
+  getCSVDownloadTypeByFilter(): 'group' | 'team' | 'user' {
+    switch (this.currentFilter) {
+      case 'Groups':
+        return 'group';
+      case 'Users':
+        return 'user';
+      case 'Teams':
+        return 'team';
+    }
+  }
+
+  onCSVExport(): void {
+    if (!this.group || !this.itemData) {
+      return;
+    }
+
+    const parentItemId = this.itemData.item.id;
+    const downloadDataType = this.getCSVDownloadTypeByFilter();
+
+    this.isCSVDataFetching = true;
+    this.progressCSVService
+      .getCSVData(this.group.id, downloadDataType, [ parentItemId ])
+      .subscribe({
+        next: data => {
+          this.isCSVDataFetching = false;
+          downloadFile([ data ], `${parentItemId}-${new Date().toDateString()}`, 'text/csv');
```

Comment:
The extension `.csv` is missing in the filename

But actually the api also provides a filename via the response header `content-disposition`
Example value: `attachment; filename=users_progress_for_group_672913018859223173_and_child_items_of_128139237432513103.csv`

I think we should use this filename, the `ProgressCSVService` could return both the csv data and the filename

Reply:
and if the header is not present, we can fallback to the filename you wrote, but with the `.csv` extension

Reply:
I guess content-disposition header is not allowed on BE side (CORS), so I can't get it on FE. @ThomasJuster @smadbe

Reply:
Case with `.csv` fixed

Reply:
It was a nice to have, but if the header is not trivial to retrieve it's alright, the feature is implemented :+1: